numpy/ma/tests/test_core.py

```diff
@@ -2581,6 +2581,13 @@ def test_no_masked_nan_warnings(self):
             # also check that allclose uses ma ufuncs, to avoid warning
             allclose(m, 0.5)
 
+    def test_masked_array_underflow(self):
+        x = np.arange(0, 3, 0.1)
+        X = np.ma.array(x)
+        with(np.errstate(under="raise")):
+            X2 = X/2.0
+            assert(X2.all() == X.all() / 2.0)
```

Comment:
```suggestion
        np.testing.assert_array_equal(X2, x/2)  
```
The `X.all()` and `X2.all()` are both False. Perhaps you wanted to write `assert np.all(X2==x/2.0)`?

Reply:
Thank you for the correction, at the time I wasn't 100% sure how I would test it but I'll change to that